specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2019-11-01/machineLearningServices.json

```diff
@@ -1978,6 +1978,146 @@
+                      "type": "string",
+                      "description": "Parameters required for this script if any."
+                    }
+                  }
+                },
+                "softwareUpdateSettings": {
+                  "type": "object",
+                  "osUpdateSettings": {
+                    "type": "object",
+                    "osUpdateType": {
+                      "type": "string",
+                      "default": "Recommended",
+                      "title": "Type of automatic operating system updates to install",
+                      "enum": [
+                        "Critical",
+                        "Recommended"
+                      ],
+                      "x-ms-enum": {
+                        "name": "osUpdateType",
+                        "modelAsString": true
+                      }
+                    },
+                    "updateFrequencyInDays": {
+                      "type": "integer",
+                      "format": "int64",
+                      "description": "Frequency of update checks and installation. Maximum allowable frequency is 30 days."
+                    },
+                    "updateHourInUtc": {
+                      "type": "integer",
+                      "format": "int64"
+                      "description": "Hour of the day (0-23) in Universal Time at which software updates can be installed, potentially requiring VM reboot."
+                    }
+                  },
+                  "sdkUpdateSettings": {
+                    "type": "object",
+                    "readonly": true,
+                    "availableUpdates": {
+                      "type": "array",
+                      "readonly": true,
+                      "description": "Describes available SDK updates for this compute instance.",
+                      "items": {
+                        "$ref": "#/definitions/AmlInstanceSdkUpdate"
+                      }
+                    }
+                  }
+                },
+                "sshSettings": {
+                  "type": "object",
+                  "sshPublicAccess": {
+                    "type": "string",
+                    "default": "Disabled",
+                    "title": "Access policy for SSH",
+                    "description": "State of the public SSH port. Possible values are: Disabled - Indicates that the public ssh port is closed on this instance. Enabled - Indicates that the public ssh port is open and accessible according to the VNet/subnet policy if applicable.",
+                    "enum": [
+                      "Enabled",
+                      "Disabled"
+                    ],
+                    "x-ms-enum": {
+                      "name": "sshPublicAccess",
+                      "modelAsString": true
+                    }
+                  },
+                  "adminCredentials": {
+                    "type": "object",
+                    "adminPublicKey": {
```

Comment:
Do you need to follow the convention to define publickey in the properties. Also as what Chris said, you need to add a readonly property for the admin name. Otherwise there is no way to tell users the admin user name.